kyo-kernel/shared/src/main/scala/kyo/kernel/ArrowEffect.scala

```diff
@@ -128,7 +128,7 @@ object ArrowEffect:
         @nowarn("msg=anonymous")
         def handleLoop(v: A < (E & S & S2), context: Context)(using Safepoint): A < (S & S2) =
             v match
-                case kyo: KyoSuspend[I, O, E, Any, A, E & S & S2] @unchecked if effectTag =:= kyo.tag =>
+                case kyo: KyoSuspend[I, O, E, Any, A, E & S & S2] @unchecked if effectTag <:< kyo.tag =>
```

Comment:
@hearnadam as discussed offline, it'd be important to check that effects with variance work well before making a release. I've added a few tests for it but they require migrating to `<:<` here. I think it's a safe since the behavior should be the same given that existing effects with variance use erased tags.